From 29d04ed11f2115232e3c21cb2b11c6a2d5788529 Mon Sep 17 00:00:00 2001
From: Kevin Wang <kevin1.wang@amd.com>
Date: Tue, 4 Jun 2019 17:41:48 +0800
Subject: [PATCH 2387/2940] drm/amd/powerplay: remove smu mutex lock in
 smu_hw_init

the smu mutex lock is unnecessary in smu hw init.

Signed-off-by: Kevin Wang <kevin1.wang@amd.com>
Reviewed-by: Evan Quan <evan.quan@amd.com>
Signed-off-by: Alex Deucher <alexander.deucher@amd.com>
---
 drivers/gpu/drm/amd/powerplay/amdgpu_smu.c | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/drivers/gpu/drm/amd/powerplay/amdgpu_smu.c b/drivers/gpu/drm/amd/powerplay/amdgpu_smu.c
index 421e1df79ae5..ad8f6dd7713d 100644
--- a/drivers/gpu/drm/amd/powerplay/amdgpu_smu.c
+++ b/drivers/gpu/drm/amd/powerplay/amdgpu_smu.c
@@ -981,8 +981,6 @@ static int smu_hw_init(void *handle)
 		}
 	}
 
-	mutex_lock(&smu->mutex);
-
 	ret = smu_feature_init_dpm(smu);
 	if (ret)
 		goto failed;
@@ -1011,8 +1009,6 @@ static int smu_hw_init(void *handle)
 	if (ret)
 		goto failed;
 
-	mutex_unlock(&smu->mutex);
-
 	if (!smu->pm_enabled)
 		adev->pm.dpm_enabled = false;
 	else
-- 
2.17.1

